### PR TITLE
Adds a Danger Process command that outputs the DSL as a JSON object

### PR DESCRIPTION
**Pull request by orta**
Re: #340 

screen shot 2017-08-18 at 07 32 44

This will now take the results of the Danger PR and pass that into the STDIN of a child process. For example this Ruby Script:

```ruby
#!/usr/bin/env ruby

puts 'Hello world'

str = STDIN.tty? ? 'Cannot read from STDIN' : $stdin.read
puts str
```

With this command `danger process ./scripts/danger_runner.rb` would echo first "Hello world" then output the JSON in the above screenshot. You could make a trivial version of Ruby Danger with this:

```ruby
#!/usr/bin/env ruby

require 'json'
dsl_json = STDIN.tty? ? 'Cannot read from STDIN' : $stdin.read
danger = JSON.parse(dsl_json)

if danger.github.pr.body.include? "Hello world"
  # TODO: sending messages back
end
```

Now I need to handle sending messages back to Danger JS.

### CODE REVIEW

**Comment by zdenektopic** [comment]
Is it somehow possible to do this for danger Ruby version and then use the output of Ruby version be parsed and worked with in danger-js?

**Comment by orta** [comment]
It's possible, as with all programming anything is possible but it's not something I plan on doing maintaining three processes that communicate with each other for this feature sounds complex. If you're looking to make Bitbucket work with Danger Process, Bitbucket support will need to be added to Danger JS.

File: changelog.md [hunk 1]
```diff
@@ -4,6 +4,42 @@
 
 // TODO
 
+* Adds a `danger process` command, this command takes amn argument of a process to run which expects the Danger DSL as JSON in STDIN,
+  and will post a DangerResults object to it's STDOUT. This frees up another process to do whatever they want. So, others
+  can make their own Danger runner.
+
+  An example of this is [Danger Swift][danger-swift]. It takes a [JSON][swift-json] file via [STDIN][swift-stdin], [compiles
```
Comment (bdotdub) [inline changelog.md]: Minor clarification: `JSON file` may be more accurate as `JSON document` if I'm understanding correctly.

Comment (orta) [reply changelog.md]: You're correct, I've just updated the changelog: https://github.com/danger/danger-js/commit/d0fc1eb6037feb68508c18e7bc0da0a603a282fb

**Pull request merged**